lib/tasks/metrics.rake

```diff
@@ -0,0 +1,46 @@
+require "prometheus/client"
+require "prometheus/client/push"
+require "prometheus/client/registry"
+require "prometheus/client/formats/text"
+
+namespace :metrics do
+  desc "Reports metrics about data in the database to prometheus via the pushgateway"
+  task report_to_prometheus: :environment do
+    default_labels = { database: "publishing-api" }
+    edition_count_dimensions = %i[state content_store document_type publishing_app locale]
+
+    prometheus_registry = Prometheus::Client::Registry.new
+    editions_in_database_gauge = prometheus_registry.gauge(
+      :editions_in_database_total,
+      docstring: "Count of editions in various databases labeled by state, document_type etc.",
+      labels: default_labels.keys + edition_count_dimensions,
+    )
+
+    edition_counts = Edition
+      .with_document
+      .where.not(state: "superseded")
+      .where.not(content_store: nil)
+      .group(*edition_count_dimensions)
+      .count
+
+    edition_counts.sort.each do |dimensions, count|
+      labels = default_labels.merge(edition_count_dimensions.zip(dimensions).to_h)
+      editions_in_database_gauge.set(count, labels:)
+    end
+
+    puts "Found #{edition_counts.count} combinations of labels"
+    puts Prometheus::Client::Formats::Text.marshal(prometheus_registry)
+
+    pushgateway_url = ENV["PROMETHEUS_PUSHGATEWAY_URL"]
+    if pushgateway_url.present?
+      puts "Pushing metrics to prometheus via #{pushgateway_url}"
+      Prometheus::Client::Push.new(
+        job: "publishing-api-metrics",
+        gateway: pushgateway_url,
+      ).add(prometheus_registry)
+    end
+  rescue StandardError => e
+    warn e.inspect
```

Comment:
Do we need this? Or is there a concern we don't log errors from rake tasks?

Reply:
First time I ran the rake task, it errored because I'd missed a `require` statement, and it didn't give me any clue what the problem was in the console logs. So I added this for debugging, saw what the error was, and then thought "we should probably log any errors we see anyway, so I'll leave the rescue in".

I was surprised rake doesn't do this by default though. Maybe there's some command line flag you need to give it for it to tell you about errors?